### PR TITLE
Bootstrapping with yeoman and adding Get and List routes

### PR DESCRIPTION
**Pull request by stjohnjohnson**
I know this is a big PR, but hear me out.

- Bootstrapped with yeoman generator
- Added Readme
- Setup both Get and List route
- Added validation for Datastore object
- 100% code coverage for plugin

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    this.datastore = datastore;
+}
+
+/**
+ * Get all builds
+ * @method getBuilds
+ * @param  {Function} callback callback function
+ */
+Handler.prototype.getBuilds = function getBuilds(callback) {
+    this.datastore.scanAll(callback);
```
Comment (jer) [inline index.js]: Is getting every single build in the system actually useful?

File: lib/get.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const GetModel = DataModel.build.get;
+const IdModel = Joi.reach(DataModel.build.base, 'id');
+
+module.exports = (handler) => ({
+    method: 'GET',
+    path: '/builds/{id}',
+    config: {
+        description: 'Get all builds',
+        notes: 'Returns all build records',
```
Comment (jer) [inline lib/get.js]: Seems inaccurate.

File: index.js (hunk 1 above)
Comment (stjohnjohnson) [inline index.js]: I totally agree, that's why I added: 
 - https://github.com/screwdriver-cd/plugin-builds/issues/4
 - https://github.com/screwdriver-cd/api/issues/5

File: lib/get.js (hunk 2 above)
Comment (stjohnjohnson) [inline lib/get.js]: Fixed

File: test/index.test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        }, (err) => {
+            if (err) {
+                throw err;
+            }
+
+            done();
+        });
+    });
+
+    afterEach(() => {
```
Comment (FenrirUnbound) [inline test/index.test.js]: Do you need to reset the stubs?

Comment (stjohnjohnson) [inline test/index.test.js]: Nah

File: lib/get.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        description: 'Get a single build',
+        notes: 'Returns a build record',
+        tags: ['api', 'builds'],
+        handler: (request, reply) => {
+            handler.getBuild(request.params.id, reply);
+        },
+        response: {
+            schema: GetModel
+        },
+        validate: {
```
Comment (nkatzman) [inline lib/get.js]: so are we moving away from using RDL, and just going to define all type validation in the `screwdriver-data-model` ?

Comment (d2lam) [inline lib/get.js]: @nkatzman  Yeah

Comment (nkatzman) [inline lib/get.js]: 👍

File: lib/get.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+const Joi = require('joi');
+const DataModel = require('screwdriver-data-model');
+
+const GetModel = DataModel.build.get;
```
Comment (tkyi) [inline lib/get.js]: Should this be a .reach() too?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,43 @@
...
+> Hapi builds plugin for the Screwdriver API
+
+## Usage
+
+```javascript
+const Hapi = require('hapi');
+const server = new Hapi.Server();
+
+server.register(require('screwdriver-plugin-builds'), {
+    config: {
```
Comment (d2lam) [inline README.md]: `options:` ?

**Comment by d2lam** [comment]
👍 
200w-2

**Comment by tkyi** [comment]
👍 
giphy-6

**Pull request merged**